### PR TITLE
rgw: log deletion status of individual objects in multi object delete request

### PR DESCRIPTION
**Pull request by cfsnyder**
Provides ops log visibility for individual objects that are deleted as part of a multi-object delete request.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

**Comment by cfsnyder** [comment]
I'm not sure why these changes would be causing the rgw_lua unit test failures:
```
expected RGW[hello] to be: NSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
```

**Comment by mattbenjamin** [comment]
@yuvalif can you check the lua aspect?

File: src/rgw/rgw_log.h [hunk 1]
```diff
@@ -13,6 +13,66 @@
 
 #define dout_subsys ceph_subsys_rgw
 
+class RGWOp;
+
+struct delete_multi_obj_entry {
+  std::string key;
+  std::string version_id;
```
Comment (mattbenjamin) [inline src/rgw/rgw_log.h:20]: put the string decls together, then http_status, then the two bool members (error and delete_marker), reorder the encode and decode to match

File: src/test/rgw/test_rgw_lua.cc [hunk 2]
```diff
@@ -169,7 +169,7 @@ TEST(TestRGWLua, EmptyScript)
 
   DEFINE_REQ_STATE;
 
-  const auto rc = lua::request::execute(nullptr, nullptr, nullptr, &s, "", script);
```
Comment (mattbenjamin) [inline src/test/rgw/test_rgw_lua.cc:172]: defer to yuval on whether this reduction of the tests is ok

**Review by mattbenjamin — COMMENTED** [review]
generally looks very sane, fix the layout of the entry struct, adding yuval to check lua behavior

**Review by yuvalif — APPROVED** [review]

File: src/rgw/rgw_log.h (hunk 1 above)
Comment (cfsnyder) [reply src/rgw/rgw_log.h]: thanks @mattbenjamin, updated!

**Review by mattbenjamin — APPROVED** [review]
lgtm

**Comment by cfsnyder** [comment]
jenkins test api

**Comment by cfsnyder** [comment]
@cbodley @mattbenjamin is there anything I need to do to move this PR along?

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
> @cbodley @mattbenjamin is there anything I need to do to move this PR along?

run teuthology against the change

**Comment by cfsnyder** [comment]
jenkins retest

**Comment by cfsnyder** [comment]
jenkins retest this please

**Comment by cfsnyder** [comment]
jenkins test make check

**Comment by cfsnyder** [comment]
jenkins test make check

**Comment by cfsnyder** [comment]
jenkins test make check

**Comment by cbodley** [comment]
@cfsnyder @ormandj it's approved, just needs qa. i'll try get a batch through today

**Pull request merged**